src/naive/datetime/mod.rs

```diff
@@ -606,6 +606,24 @@ impl NaiveDateTime {
         Some(NaiveDateTime { date, time })
     }
 
+    /// Add a duration in [`Days`] to the date part of the `NaiveDateTime`
+    ///
+    /// Returns `None` if the resulting date would be out of range.
+    pub fn checked_add_days(self, days: Days) -> Option<Self> {
+        let new_date = self.date().checked_add_days(days)?;
```

Comment:
Simplifying:

```rust
Some(Self {
    date: self.date.checked_add_days(days)?,
    time: self.time,
})
```

Reply:
good call - I've also used the `..self` syntax, although I can use the above if you prefer for consistency